test/Dialect/FIRRTL/errors.mlir

```diff
@@ -1176,6 +1176,30 @@ firrtl.circuit "ForceableTypeMismatch" {
   }
 }
 
+// -----
+
+firrtl.circuit "RefForceProbe" {
+  firrtl.module @RefForceProbe() {
+    %a = firrtl.wire : !firrtl.uint<1>
+    %1 = firrtl.ref.send %a : !firrtl.uint<1>
+    // expected-note @above {{prior use here}}
+    // expected-error @below {{use of value '%1' expects different type than prior uses: '!firrtl.rwprobe<uint<1>>' vs '!firrtl.probe<uint<1>>'}}
```

Comment:
Thanks for adding tests for this. IMHO it's fine to not add error checks for things ODS guarantees, but no problem having more checks.

Reply:
Yeah, this is less about verifying the details of the diagnostic as much as ensuring /something/ catches this and that things don't crash before producing some sort of relevant message.